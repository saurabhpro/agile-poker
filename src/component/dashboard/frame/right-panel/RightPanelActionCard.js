import React from 'react';
import { Card, Button } from 'react-bootstrap';
import PropTypes from 'prop-types';

import './RighSideCard.css';
import { removeTasksOfTeam } from '../../../utils/firebaseDb';
import TaskIdManager from './TaskIdManager';

const RightPanelActionCard = ({ currentUser, style }) => {
  const [showReset, setShowReset] = React.useState(false);

  const reset = () => {
    removeTasksOfTeam(currentUser.team, '')
      .then(() => {
        window.location.href = `${process.env.PUBLIC_URL}/`;
        console.log('Document successfully deleted!');
      })
      .catch((error) => {
        console.error('Error removing document: ', error);
      });
  };

  return (
    <Card className="rightCard" style={style}>
      <Card.Body>
        <Card.Title className="title">Poker</Card.Title>

        <Card.Text className="text">{currentUser.userName}</Card.Text>

        <p className="text">
          {currentUser.role
            ? currentUser.role.toUpperCase()
            : 'TEAM MEMBER'}
        </p>

        <TaskIdManager
          team={currentUser.team}
          role={currentUser.role.toUpperCase()}
          setShowReset={setShowReset}
        />

        {showReset &&
          currentUser.role.toUpperCase() !== 'TEAM MEMBER' && (
            <div>
              {' '}
              <Button block onClick={() => reset()} variant="light">
                Reset
              </Button>
            </div>
          )}
      </Card.Body>
    </Card>
  );
};

RightPanelActionCard.propTypes = {
  currentUser: PropTypes.object.isRequired,
  style: PropTypes.object.isRequired,
};

export default RightPanelActionCard;
