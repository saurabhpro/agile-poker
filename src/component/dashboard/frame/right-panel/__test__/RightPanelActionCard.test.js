import React from 'react';
import { shallow } from 'enzyme';
import RightPanelActionCard from '../RightPanelActionCard';

let component;

const user = {
  userName: 'Saurabh',
  isOnline: false,
  isActive: false,
  memberSince: new Date(2020, 20, 8),
  team: 'r2unit',
  role: 'Scrum Master',
};

describe('result from poker card panel', () => {
  beforeAll(() => {
    component = shallow(
      <RightPanelActionCard
        currentUser={user}
        style={{ border: 'none' }}
      />,
    );
    //console.debug("This was prited",component.debug());
  });

  it('matches the rendered snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  it('should render one <Card>', () => {
    expect(component.find('Card')).toHaveLength(1);
  });
});
