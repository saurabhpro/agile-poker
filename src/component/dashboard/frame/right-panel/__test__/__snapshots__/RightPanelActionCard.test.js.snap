// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`result from poker card panel matches the rendered snapshot 1`] = `
<Card
  body={false}
  className="rightCard"
  style={
    Object {
      "border": "none",
    }
  }
>
  <CardBody>
    <CardTitle
      className="title"
    >
      Poker
    </CardTitle>
    <CardText
      className="text"
    >
      SAURABH
    </CardText>
    <p
      className="text"
    >
      SCRUM MASTER
    </p>
    <Button
      active={false}
      block={true}
      disabled={false}
      onClick={[Function]}
      variant="Link"
    >
      Reset
    </Button>
  </CardBody>
</Card>
`;
